### DISCUSSION TITLE
Creating a game controller

### DISCUSSION THREAD

**Discussion by KnoblauchSuppe** [NicoHood/HID#256] [Q&A]
@NicoHood  Hello, thanks for the awesome project, to allow non 32u4 chips to be used as HID devices. I am trying to create a flight simulator controller, or more accurately a flight stick. I would need 6 axis and some buttons (about 10,not much). Is it possible to use the arduino joystick library? (this library: https://github.com/MHeironimus/ArduinoJoystickLibrary ) Id like to know if its possible to do it before modifying my arduino. I am using an arduino UNO, which has 6 analog inputs. I would create something like the attached file

![Screenshot 2021-02-25 115954](https://user-images.githubusercontent.com/75313906/109144077-04250480-7761-11eb-9a82-027e1541e03a.png)
Thanks in advance

**Comment by NicoHood** [comment]
I dont know the other library, but why not? You could also use the Gamepad of this library. Gamepad, Joystick is just the same.

But if you are using the 16u2 of the Uno, you need to create a serial communication between the two MCUs. This can be tricky. I recommend using a 32u4 here.

**Reply by KnoblauchSuppe** [reply]
Yes, i wanted to get a 32u4 and i understand its better but the current shipping times are very long here, thats why id like to use a 16u2 as a replacement. Is it possible to just have 4 axis and a couple of buttons without creating the serial communication? Or is there any way easier, since im not that experienced with arduinos and computers
Edit: i read something last time about loading a custom hex file on the arduino to enable it to be used as a HID device, is that the serial communication you were talking about? And i will use the gamepad library, since they look like they have the same functionality

**Reply by NicoHood** [reply]
Checkout this:
https://github.com/NicoHood/HoodLoader2

And read the Wiki carefully. But the 16u2 has not analog pins exposed:
https://github.com/NicoHood/HoodLoader2/wiki/Arduino-Uno-Mega-16u2-Pinout

**Reply by KnoblauchSuppe** [reply]
Oh i see, only 7 i/o pins. Thanks for the reply, will order a leonardo or a micro then, thanks for the help, have a good day.
